lib/rules/quotes.js

```diff
@@ -123,12 +123,26 @@ module.exports = {
 
         /**
          * Determines if a given node is part of JSX syntax.
```

Comment:
I think shouldn't.
I thought `Literal`s of `JSXAttribute`/`JSXElement` is part of JSX syntax. Those are `JSXDoubleStringCharacters`/`JSXSingleStringCharacters`/`JSXText` production in original spec. On the other hand, inside of braces is `AssignmentExpression` production of ECMAScript.